### PR TITLE
499 Inconsistent padding

### LINKED ISSUE #499: Inconsistent padding
# :clipboard: Bug Details

>Inconsistent padding 

key | value
--|--
Reported At | 2024-11-08 10:18:10 UTC
Email | jeffrey.doyle@flowfoundation.org
Categories | Suggest an improvement
App Version | 2.2.9 (10)
Session Duration | 75
Device | iPhone 13 Pro, iOS 17.6.1
Display | 390x844 (@3x)
Location | Bangkok, Thailand (en-CA)

## :point_right: [View Full Bug Report on Instabug](https://dashboard.instabug.com/applications/flow-foundation/production/bugs/137?utm_source=github&utm_medium=integrations) :point_left:
___
# :chart_with_downwards_trend: Session Profiler
Here is what the app was doing right before the bug was reported:

Key | Value
--|--
CPU Load | 48.7%
Used Memory | 7.0% - 0.21/3.0 GB
Used Storage | 97.9% - 233.3/238.3 GB
Connectivity | NA - NA
Battery | 30% - plugged
Orientation | portrait

Find all the changes that happened in the parameters mentioned above during the last 60 seconds before the bug was reported here:  :point_right: **[View Full Session Profiler](https://dashboard.instabug.com/applications/flow-foundation/production/bugs/137?show-session-profiler=true&utm_source=github&utm_medium=integrations)** :point_left:
___
# :mag_right: Logs
### Network Log
This bug report has **2 failed** (4xx, 5xx or no response) requests. Here are some of them:
```
10:13:14 400 POST https://rest-mainnet.onflow.org/v1/scripts?block_height=sealed
10:17:17 400 POST https://rest-mainnet.onflow.org/v1/scripts?block_height=sealed

```
Find all the network requests with more details about each one, its duration, method, URL, headers, and response here:  :point_right: **[View Detailed Network Logs](https://dashboard.instabug.com/applications/flow-foundation/production/bugs/137?show-logs=network_log&utm_source=github&utm_medium=integrations)** :point_left:

___
# :camera: Images
[image attachment](https://d38gnqwzxziyyy.cloudfront.net/attachments/bugs/23226997/2e41ee9e10d7d34720fc66fd07bb5408_original/36890604/2024110805180315857782.jpg?Expires=4886734697&Signature=J4gFk6cg4A052lGYgDdXLhF7xJ3JverUhZ7zsRSE45VbSCLbAtYNdwbF2wPy~ljPKIMQ8H-66n9GbyoxhUXti51Ayeqr3BMW3I8TSKmRiKo2mNgTRAgL0VWjO8rCuLGVRvDA-u~2wUfJmJSxsOKLCtn-XZqMgBXkgGpztXqOXRhTMol-~RhOXwr13v3I6zIDY37JIzn-CWorFGtvKsc2ZgD8KIJdP53TgU6FY24PS4x96MpDEBffE8sfUM0A0q-0A3isbsXVreu3n02yYtDi7~UMoRtL7ancNLyWo19B-JOwSJW3nBjsFT0X5Ev4Gljh5iHciwlZKClgKPeNKWfzYA__&Key-Pair-Id=APKAIXAG65U6UUX7JAQQ)

___
# :warning: Looking for More Details?
1. **User Steps**: [**enable them**](https://docs.instabug.com/docs/ios-logging?utm_source=github&utm_medium=integrations#section-user-steps) to get the most out of your plan and see every action users take before reporting a bug.
2. **User Events**: start capturing custom User Events to send them along with each report. [**Find all the details in the docs**](https://docs.instabug.com/docs/ios-logging?utm_source=github&utm_medium=integrations).
3. **Instabug Log**: start adding Instabug logs to see them right inside each report you receive. [**Find all the details in the docs**](https://docs.instabug.com/docs/ios-logging?utm_source=github&utm_medium=integrations).
4. **Console Log**: when enabled you will see them right inside each report you receive. [**Find all the details in the docs**](https://docs.instabug.com/docs/ios-logging?utm_source=github&utm_medium=integrations).

### PR DESCRIPTION
**Pull request by jeden**
https://github.com/Outblock/FRW-iOS/issues/499

### CODE REVIEW

File: FRW/UI/Component/HalfSheetModal.swift [hunk 1]
```diff
@@ -60,6 +60,25 @@ struct SheetHeaderView: View {
...
 
+@available(iOS 16, *)
+struct HalfSheetModifier: ViewModifier {
+    @State var height: CGFloat = 0
+    
+    func body(content: Content) -> some View {
+        content
+            .fixedSize(horizontal: false, vertical: true)
+            .getHeight($height)
+            .presentationDetents([.height(height), .large])
```
Comment (jeden) [inline FRW/UI/Component/HalfSheetModal.swift:71]: `height()` is available on iOS 16 only, that's the reason why this fix doesn't work on older versions

**Comment by lmcmz** [comment]
Have you tested your approach with more UI context ?
I doesn't look work for me
We need a solution which can fit the UI context for the bottom sheet dynamically.

And also the gap need to be handled

IMG_4540

IMG_4539

**Comment by jeden** [comment]
> Have you tested your approach with more UI context ? I doesn't look work for me We need a solution which can fit the UI context for the bottom sheet dynamically.
> 
> And also the gap need to be handled
> 

@lmcmz 
I did test it with some custom content I duplicated from (via copy and paste), as well as increasing the font size. Also I checked on a few devices (including the iPhone X and SE). I'll double check that again.

I think I didn't test with the keyboard though, I'll check that padding issue — it looks like a safe area padding issue

Just to be sure, you are at least on iOS 16, right?

**Comment by lmcmz** [comment]
Another thing is when the popup is expandable, the solution will broke the UI
IMG_4543

**Comment by jeden** [comment]
image

**Comment by jeden** [comment]
image

image

image

**Comment by jeden** [comment]
image

image

image

**Pull request closed without merging**